app/uk/gov/hmrc/devhubsupportfrontend/services/AuditService.scala

```diff
@@ -28,7 +28,47 @@ import uk.gov.hmrc.devhubsupportfrontend.connectors.ApiPlatformDeskproConnector
 @Singleton
 class AuditService @Inject() (val auditConnector: AuditConnector)(implicit ec: ExecutionContext) {
 
-  def explicitAudit(auditType: String, ticket: ApiPlatformDeskproConnector.CreateTicketRequest)(implicit hc: HeaderCarrier): Unit = {
-    auditConnector.sendExplicitAudit(auditType, Json.toJson(ticket))
+  def explicitAudit(auditAction: AuditAction)(implicit hc: HeaderCarrier): Unit = {
+    val details = auditAction match {
+      case action: CreateTicketAuditAction => Json.toJson(CreateTicketDetails(action.createTicketRequest))
+    }
+    auditConnector.sendExplicitAudit(auditAction.auditType, details)
   }
 }
+
+sealed trait AuditAction {
```

Comment:
I think this and it's one (currently) implementation should be broken out into their own file.